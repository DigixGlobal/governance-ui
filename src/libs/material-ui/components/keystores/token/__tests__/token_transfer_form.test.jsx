import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import TokenTransferForm from '../token_transfer_form';

const formData = {
  from: '0x12345',
};

const contract = {
  balanceOf: () => ({
    shift: () => ({}),
  }),
};

const token = {
  color: 'blue',
  symbol: 'GBP',
  decimals: 2,
};

describe('<TokenTransferForm />', () => {
  test('renders correctly', () => {
    const component = shallow(
      <TokenTransferForm
        formChange={jest.fn}
        formData={formData}
        contract={contract}
        token={token}
      />,
    );

    expect(toJson(component)).toMatchSnapshot();
  });
});
