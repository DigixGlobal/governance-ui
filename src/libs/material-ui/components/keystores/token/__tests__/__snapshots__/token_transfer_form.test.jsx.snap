// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TokenTransferForm /> renders correctly 1`] = `
<FormField>
  <AddressInput
    formChange={[Function]}
    formData={
      Object {
        "from": "0x12345",
      }
    }
    hideDropdown={false}
    label="From"
    name="from"
    placeholder="e.g. \`0x123...456\`"
    showQrScanner={false}
  />
  <AddressInput
    formChange={[Function]}
    formData={
      Object {
        "from": "0x12345",
      }
    }
    hideDropdown={false}
    label="To"
    name="to"
    placeholder="e.g. \`0x123...456\`"
    showQrScanner={true}
  />
  <ValueInput
    color="blue"
    formChange={[Function]}
    formData={
      Object {
        "from": "0x12345",
      }
    }
    label="Number of Tokens"
    name="tokenValue"
    sendAll={Object {}}
    symbol="GBP"
  />
</FormField>
`;
